src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/listeners/DoCleanupListener.groovy

```diff
@@ -29,5 +47,26 @@ class DoCleanupListener extends AbstractSpringListener {
                 }
             }
         }
+        if (ISL_PROPS_DB_RESET in error.method.getAnnotation(Tags).value()) {
+            def isls = topology.getIslsForActiveSwitches().collectMany { [it, it.reversed] }
+            log.info("Resetting ISLs bandwidth due to the failure in " + error.method.parent.name + "\nISLs: " + isls)
+            isls.each { database.resetIslBandwidth(it) }
+            log.info("Resetting ISLs cost due to the failure in " + error.method.parent.name)
+            database.resetCosts(topology.isls)
+        }
+
+        if (ISL_RECOVER_ON_FAIL in error.method.getAnnotation(Tags).value()) {
+            def failedLinks = northbound.getAllLinks().findAll { it.state == FAILED }
+            failedLinks.collectMany { return [it.source.switchId, it.destination.switchId] }.unique()
+                    .each { switchId ->
+                        northbound.getPorts(switchId).findAll { port -> port.config == ["PORT_DOWN"] }
+                                .each { portDetails -> antiflap.portUp(switchId, portDetails.portNumber) }
+                    }
+            if (failedLinks) {
+                wait(WAIT_OFFSET * 2) {
```

Comment:
There's the specialized timeout value for links discovery, something like 'toplologyDiscovery'. Please, use the one.

Reply:
done